### PR TITLE
ADR-1097 Add no AppaId page

### PR DESCRIPTION
**Pull request by davidfes**

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -84,14 +84,16 @@ mongodb {
 }
 
 urls {
-  login         = "http://localhost:9949/auth-login-stub/gg-sign-in"
-  loginContinue = "http://localhost:16000/manage-alcohol-duty"
-  signOut       = "http://localhost:9025/gg/sign-out"
+  login                = "http://localhost:9949/auth-login-stub/gg-sign-in"
+  loginContinue        = "http://localhost:16000/manage-alcohol-duty"
+  signOut              = "http://localhost:9025/gg/sign-out"
+  appaIdRegister       = "http://localhost:16000/manage-alcohol-duty"
```
Comment (davidfes) [inline conf/application.conf:90]: To be updated when the guidance is ready

File: app/controllers/actions/IndentifierWithoutEnrolment.scala [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  override def invokeBlock[A](
+    request: Request[A],
+    block: IdentifierWithoutEnrolmentRequest[A] => Future[Result]
+  ): Future[Result] = {
+
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
+
+    authorised(predicate).retrieve(internalId and groupIdentifier) { case optInternalId ~ optGroupId =>
+      val internalId: String = getOrElseFailWithUnauthorised(optInternalId, "Unable to retrieve internalId")
```
Comment (yasaracar) [inline app/controllers/actions/IndentifierWithoutEnrolment.scala:65]: I think we should have kept this simple, we don't need IdentifierWithoutEnrolmentRequest for these pages. I can't foresee any resaon we'd use internalId and groupId apart from logging but happy if it stays as well.

File: app/controllers/actions/IndentifierWithoutEnrolment.scala [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      val groupId: String    = getOrElseFailWithUnauthorised(optGroupId, "Unable to retrieve groupIdentifier")
+      block(IdentifierWithoutEnrolmentRequest(request, groupId, internalId))
+    } recover {
+      case _: NoActiveSession        =>
```
Comment (yasaracar) [inline app/controllers/actions/IndentifierWithoutEnrolment.scala:69]: No active session is an AuthorisationException as well so I think it makes more sense that it is in handleAuthException

File: test/controllers/actions/IndentifierWithoutEnrolmentActionSpec.scala [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        )(any(), any())
+      )
+        .thenReturn(Future(new ~(Some(internalId), None)))
+
+      intercept[UnauthorizedException] {
+        await(identifierAction.invokeBlock(FakeRequest(), testAction))
+      }
+    }
+
+    "redirect to the authorised page if not be authorised" in {
```
Comment (yasaracar) [inline test/controllers/actions/IndentifierWithoutEnrolmentActionSpec.scala:123]: should be unauthorised page

File: test/controllers/actions/IndentifierWithoutEnrolmentActionSpec.scala [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+      List(
+        InsufficientEnrolments(),
+        InsufficientConfidenceLevel(),
+        UnsupportedAuthProvider(),
+        UnsupportedAffinityGroup(),
+        UnsupportedCredentialRole(),
+        IncorrectCredentialStrength()
+      ).foreach { exception =>
```
Comment (yasaracar) [inline test/controllers/actions/IndentifierWithoutEnrolmentActionSpec.scala:131]: Add UnauthorizedException as well please

**Review by yasaracar — CHANGES_REQUESTED** [review]

File: app/controllers/actions/IndentifierWithoutEnrolment.scala (hunk 3 above)
Comment (davidfes) [reply app/controllers/actions/IndentifierWithoutEnrolment.scala:69]: It's the same as the catch all, so yes.

File: app/controllers/actions/IndentifierWithoutEnrolment.scala (hunk 2 above)
Comment (davidfes) [reply app/controllers/actions/IndentifierWithoutEnrolment.scala:65]: Given it's a small number of pages I don't mind either way. Your call.

File: test/controllers/actions/IndentifierWithoutEnrolmentActionSpec.scala (hunk 4 above)
Comment (davidfes) [reply test/controllers/actions/IndentifierWithoutEnrolmentActionSpec.scala:123]: Fixed, and 'if not be' sounds a bit pirate to me. Yay!

File: conf/messages.en [hunk 6]
```diff
@@ -69,6 +70,12 @@ signedOut.guidance = We did not save your answers.
 unauthorised.title = You can’t access this service with this account
 unauthorised.heading = You can’t access this service with this account
 
+noAppaId.title = You need an Alcoholic Products Producer approval ID to access Alcohol Duty
+noAppaId.heading = You need an Alcoholic Products Producer approval ID to access Alcohol Duty
+noAppaId.text = Register with HMRC to receive an Alcoholic Products Producer approval ID and manage your alcohol duty online.
+noAppaId.link.id = register-for-alcohol-duty
```
[the remark this answers is not in the document]
Comment (davidfes) [reply conf/messages.en:76]: Also was broken as its use wasn't referring to messages.

**Review by yasaracar — APPROVED** [review]

**Review by yasaracar — APPROVED** [review]

**Pull request merged**